python/cudf/cudf/core/index.py

```diff
@@ -2142,6 +2141,8 @@ def __init__(
         if yearfirst is not False:
             raise NotImplementedError("yearfirst == True is not yet supported")
 
+        self._freq = _validate_freq(freq)
```

Comment:
While looking on adding `freq` support before, I found some APIs manipulate `freq`(to new values) and return new results. (_I vaguely remember..but I think that happens in binops?_) Should we add a TODO comment here that this is not fully functional yet and `freq` support needs to be added in rest of the code-base?

Reply:
Yes, although maybe the default behaviour could be for `DatetimeIndex` to infer `freq` from its values. Then this should just work.

Also, we should probably only do that in compatibility mode for perf reasons.